src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -514,9 +429,51 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Intrinsified(byte* pBuff
             // We can't perform vectorized reads at this point, so we'll fall back to reading primitives
             // directly. Note that all of these reads are unaligned.
 
-            Debug.Assert(bufferLength < SizeOfVector128);
+            Debug.Assert(bufferLength < SizeOfVector256);
 
-            // QWORD drain
+            // 1st and 2nd QWORD drain
+
+            if ((bufferLength & 16) != 0)
+            {
+                if (UIntPtr.Size == sizeof(ulong))
```

Comment:
It's not clear to me why `64-bit tzcnt` is limited to 64-bit platforms.

On 32-bit this will "at worst" be a single compare against zero and a `bsf` which I'd expect should be faster than two `AllBytesInUInt32AreAscii` checks?